### PR TITLE
Activity fixes

### LINKED ISSUE #3337: Missing info in activities after running integration for couple of hours
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ x] Bug report  
[ ] Documentation issue or request
</code></pre>

## The problem
I have scenario consist of 6 integration with 3messages/s in total. After couple of hours of running (~4 hours 60000 messages)  I'm missing steps infromations in activities:
No steps:
screen shot 2018-08-07 at 16 39 02
screen shot 2018-08-07 at 16 42 34

## Expected behavior

always should be all steps visible: 
screen shot 2018-08-07 at 16 38 53

## Tasks involved / Steps to Reproduce

1.
2.
3.
4.

### PR DESCRIPTION
**Pull request by zregvart**
I found a lot of issues with activities, please review thoroughly, details are in the commit messages.

Fundamental change is that we no longer keep 1 days worth of activities, but 50 entries (by default).

Fixes #3337

### CODE REVIEW

**Comment by zregvart** [comment]
I've been running integration that logs every second, during 3 hours no buildup of threads, PostgreSQL /var/lib/pgsql/data/userdata is at 71MB, I see the most recent activities and cleanup thread deletes ~900 activities (as expected) every 15mins.

Looking good.

**Comment by zregvart** [comment]
In logs in /var/lib/pgsql/data/userdata/pg_log I can see autovacuum and analyse working, they don't have timestamps so I'm not sure how often are they run:
```
LOG:  automatic vacuum of table "syndesis.public.jsondb": index scans: 1
	pages: 0 removed, 151 remain, 0 skipped due to pins
	tuples: 882 removed, 6114 remain, 0 are dead but not yet removable
	buffer usage: 922 hits, 0 misses, 23 dirtied
	avg read rate: 0.000 MB/s, avg write rate: 6.561 MB/s
	system usage: CPU 0.00s/0.00u sec elapsed 0.02 sec
LOG:  automatic analyze of table "syndesis.public.jsondb" system usage: CPU 0.00s/0.10u sec elapsed 0.10 sec
```

**Comment by zregvart** [comment]
The 1sec to log integration is running now for 19 hours, I see all activities `/var/lib/pgsql/data/userdata/` is at 72MB, no buildup of threads.

I have another integration that logs every 40min (35min is the set socket read timeout), this one recovers from the timeout, and continues to gather activities:

```
2018-08-10 07:20:27.314  INFO [-,,,] 1 --- [5:27.299608000Z] i.s.s.l.j.c.ActivityTrackingController   : Streaming ended for pod i-40mintolog-2-ns28z due to: java.net.SocketTimeoutException: timeout, caused by: java.net.SocketException: Socket closed
2018-08-10 07:20:28.816  INFO [-,,,] 1 --- [ning]: pollPods] i.s.s.l.j.c.ActivityTrackingController   : Recovered state: {"time":"2018-08-10T06:45:27.301478000Z"}
2018-08-10 07:20:28.822  INFO [-,,,] 1 --- [intolog-2-ns28z] i.s.s.l.j.c.ActivityTrackingController   : Getting controller for pod: i-40mintolog-2-ns28z
```

This is looking really good.

**Comment by zregvart** [comment]
Seems that review is not going to happen, I feel confidant in this after running the integration for a while, so I'm merging it.

**Comment by rhuss** [comment]
👍

File: app/server/logging/jsondb/src/main/java/io/syndesis/server/logging/jsondb/controller/ActivityTrackingController.java [hunk 1]
```diff
@@ -365,7 +365,12 @@ private void processEventQueue() {
...
                     // Write the batch..
-                    writeBatch(batch);
+                    try {
+                        writeBatch(batch);
+                    } catch (@SuppressWarnings("PMD.AvoidCatchingGenericException") RuntimeException e) {
+                        LOG.warn("Unable to write batch of events: ", e.getMessage());
+                        LOG.debug("Unable to write batch of events: ", e);
+                    }
 
                     LOG.debug("Batch ingested {} log events", eventCounter);
```
Comment (chirino) [inline app/server/logging/jsondb/src/main/java/io/syndesis/server/logging/jsondb/controller/ActivityTrackingController.java]: We should not log this message on failure.  Perhaps move it inside the try?

File: app/server/logging/jsondb/src/main/java/io/syndesis/server/logging/jsondb/controller/KubernetesSupport.java [hunk 2]
```diff
@@ -69,7 +69,7 @@ protected void watchLog(String podName, Consumer<InputStream> handler, String si
                 .append(pod.getResourceUrl().toString())
                 .append("/log?pretty=false&follow=true&timestamps=true");
             if (sinceTime != null) {
-                url.append("&sinceTime=");
+                url.append("&sinceTime=").append(sinceTime);
```
Comment (chirino) [inline app/server/logging/jsondb/src/main/java/io/syndesis/server/logging/jsondb/controller/KubernetesSupport.java]: Ah.. less load on reconnect.. good catch.

File: app/server/logging/jsondb/src/main/java/io/syndesis/server/logging/jsondb/controller/KubernetesSupport.java [hunk 3]
```diff
@@ -91,9 +91,7 @@ public void onResponse(final Call call, final Response response) throws IOExcept
                         Thread.currentThread().setName("Logs Controller [running], streaming: " + podLogUrl);
                         try {
                             if (response.code() == 200) {
-                                try (InputStream responseStream = response.body().byteStream()) {
```
Comment (chirino) [inline app/server/logging/jsondb/src/main/java/io/syndesis/server/logging/jsondb/controller/KubernetesSupport.java]: I trust you verified we don't run into a resource leak right?

**Pull request merged**